<?xml version="1.0" encoding="UTF-8"?>
<UI version="4.0">
 <class>OpenAISettingsClass</class>
 <widget class="QDialog" name="OpenAISettingsClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Manage OpenAI Accounts</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_main">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_content">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_list">
       <item>
        <widget class="QListWidget" name="listAccounts"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_listBtns">
         <item>
          <widget class="QPushButton" name="btnAddAccount">
           <property name="text">
            <string>+</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btnRemoveAccount">
           <property name="text">
            <string>-</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QGroupBox" name="groupDetails">
       <property name="title">
        <string>Account Details</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_details">
        <item>
         <widget class="QLabel" name="label_name">
          <property name="text">
           <string>Display Name:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="editName"/>
        </item>
        <item>
         <widget class="QCheckBox" name="checkAzure">
          <property name="text">
           <string>Use Azure OpenAI</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_key">
          <property name="text">
           <string>API Key:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="editKey">
          <property name="echoMode">
           <enum>QLineEdit::Password</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="label_model">
          <property name="text">
           <string>Model / Deployment:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="editModel"/>
        </item>
        <item>
         <widget class="QLabel" name="label_prompt">
          <property name="text">
           <string>Default System Prompt:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="editPrompt"/>
        </item>
        <item>
         <widget class="QLabel" name="label_url">
          <property name="text">
           <string>Base URL / Full Endpoint:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="editUrl"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_test">
          <item>
           <widget class="QPushButton" name="btnTest">
            <property name="text">
             <string>Test Connection</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_test">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="labelHelp">
          <property name="styleSheet">
           <string notr="true">font-size: 10px;</string>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>OpenAISettingsClass</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>OpenAISettingsClass</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</UI>